dyne/examples.py

```diff
@@ -533,3 +534,155 @@ def h(k, X, with_jacobian=True):
 
     return NonlinearProblemExample(X0, P0, f, Q, [(measurement_epochs, Z, h, R)],
                                    n_epochs, Xt, Wt)
+
+
+def generate_magnetic_heading(
+    total_time=120,
+    time_step=0.1,
+    rph_mean=[3, -5, 0],
+    rph_change_amplitude=5,
+    rph_change_period=[5, 5, 30],
+    rph_change_phase_offset=[0, 90, 45],
+    magnetic_field = [10, 3, 40],
+    mag_bias=[30, -20, 70],
+    X0=None,
+    P0=np.diag([180, 50, 50, 50])**2,
+    q=np.zeros(4),
+    sigma_measurement=[1, 1, 1],
+    rng=0
+):
+    """Generate data for exapmle of heading estimation using biased magnetometr data.
+
+    Roll, pitch and heading angles are varying according to a given harmonic law.
+    Exact values of roll and pitch are known, heading and magnetometr biases
+    should be estimated.
+
+    The discrette time system model is::
+
+    h[k+1] = h[k] + dh[k] + w_h[k]
+    bx[k+1] = bx[k] + w_x[k]
+    by[k+1] = by[k] + w_y[k]
+    bz[k+1] = bz[k] + w_z[k]
+
+    The measurement model is::
+
+    [x, y, z] = C_bn @ [mx, my, mz] + [bx, by, bz]
+
+    Measurement of ``x``, ``y`` and ``z`` components of magnetic field are avaliable.
+
+
+    Parameters
+    ----------
+    total_time : float
+        Total time of the simulation.
+    time_step : float
+        Discretization time step.
+    rph_mean: array_like, shape(3,)
+        Mean value of angles in degrees.
+    rph_change_amplitude: float or array_like, shape(3,)
+        Amplitude of angles variation in degrees.
+    rph_change_period: float or array_like, shape(3,)
+        Period of sinusoidal angles change in seconds.
+    rph_change_phase_offset: array_like, shape(3,)
+        Phase offset for sinusoid part in degrees.
+    magnetic_field: array_like, shape(3,)
```

Comment:
So basically it's constant, now I see.

Also I think adding frame index like "magnetic_field_n" will be useful.